Document the new skipped response parameter of the _bulk_action endpoint (#2731)
elastic/security-docs: docs/detections/api/rules/rules-api-bulk-actions.asciidoc

[discrete]
===== Response payload

For `enable`, `disable`, `delete`, `edit`, and `duplicate` actions, a JSON object containing the action's outcome:

> - `attributes.summary.total`: total number of rules matching the bulk action
> - `attributes.summary.succeeded`: number of successful outcomes (number of rules that were enabled or deleted or updated - depending on the type of action)
> - `attributes.summary.failed`: number of failed outcomes
> - `attributes.summary.skipped`: number of rules that were skipped due to various reasons (see below)
> - `attributes.results.created`: rule objects that were created during the action's execution
> - `attributes.results.updated`: rule objects that were updated during the action's execution
> - `attributes.results.deleted`: rule objects that were deleted during the action's execution
> - `attributes.results.skipped`: rules that were skipped during the action's execution

Reviewer: Added a few small style edits, and a revision to tighten up the description (it's probably clear enough that "enabled, deleted, or updated" are different types of actions, without needing to point that out).
```suggestion
- `attributes.summary.total`: Total number of rules matching the bulk action
- `attributes.summary.succeeded`: Number of successful outcomes (number of rules that were enabled, deleted, or updated)
- `attributes.summary.failed`: Number of failed outcomes
- `attributes.summary.skipped`: Number of rules that were skipped due to various reasons (explained below)
- `attributes.results.created`: Rule objects that were created during the action's execution
- `attributes.results.updated`: Rule objects that were updated during the action's execution
- `attributes.results.deleted`: Rule objects that were deleted during the action's execution
- `attributes.results.skipped`: Rules that were skipped during the action's execution
```

[discrete]
===== Response payload

> For `enable`, `disable`, `delete`, `edit`, and `duplicate` actions, a JSON object containing the action's outcome: number of failed and successful outcomes, number of rules, and number of rule objects created, deleted, skipped or updated (depending on the action).

Reviewer 2: `number of rules, and number of rule objects created, deleted, skipped, or updated (depending on the action)`

It's not the number of rule objects, we return the objects themselves.

What about this:

```
For `enable`, `disable`, `delete`, `edit`, and `duplicate` actions, a JSON object containing the action's outcome:

- `attributes.summary.total`: total number of rules matching by the bulk action.
- `attributes.summary. succeeded`: number of successful outcomes (number of rules that were enabled or deleted or updated - depending on the type of action)
- `attributes.summary.failed`: number of failed outcomes
- `attributes.summary.skipped`: number of rules that were skipped due to various reasons (see below).
 number of failed and successful outcomes, number of rules, and number of rule objects created, deleted, skipped, or updated (depending on the action)
- `attributes.results.created`: rule objects that were created during the action's execution
- `attributes.results.updated`: rule objects that were updated during the action's execution
- `attributes.results.deleted`: rule objects that were deleted during the action's execution
- `attributes.results.skipped`: rules that were skipped during the action's execution
```

> For `enable`, `disable`, `delete`, `edit`, and `duplicate` actions, a JSON object containing the action's outcome: number of failed and successful outcomes, number of rules, and number of rule objects created, deleted, skipped or updated (depending on the action).

Reviewer: @banderror This looks good, just added a couple small edits to the bulleted list [here](https://github.com/elastic/security-docs/pull/2731/files#r1060888603). Thank you!

> Rules can only be `skipped` when performing an  `edit` action, when all the operations to be performed on any one rule result in a no-op. For example, if the `edit` action is to add a tag to a rule that already has that tag, and to delete an index pattern that doesn't exist. Skipped rules objects will only return their `id`, `name`, and `skip_reason`.

Reviewer: "No-op" is a bit of jargon, so this edit tries a simpler explanation — please confirm if it's accurate, as it's a new term for me.

Also a little uncertain about "and" or "or" with the two examples. I _think_ they're just two separate examples, so "or" would make sense, but if the two examples are actually connected and have to function as a combination of actions, then "and" is probably best.
```suggestion
Rules can only be `skipped` when performing an `edit` action, when all the operations to be performed on any one rule result in nothing being done. For example, if the `edit` action is to add a tag to a rule that already has that tag, or to delete an index pattern that doesn't exist. Skipped rules objects will only return their `id`, `name`, and `skip_reason`.
```

Reviewer 2: I agree with @joepeeples suggestions regarding the noop term and the "and vs or". I think "or" would be easier to grasp for the user. @joepeeples FYI it could be "and" as well because the API supports sending multiple bulk edit operations in a single request.

I'd simplify it a bit further:

```suggestion
A rule can only be `skipped` when the bulk action to be performed on it results in nothing being done. For example, if the `edit` action is used to add a tag to a rule that already has that tag, or to delete an index pattern that is not specified in a rule. Objects returned in `attributes.results.skipped` will only include rules' `id`, `name`, and `skip_reason`.
```

Not mentioning the `Rules can only be skipped when performing an edit action` gives us the freedom to expand skipping to other cases.

> Rules can only be `skipped` when performing an  `edit` action, when all the operations to be performed on any one rule result in a no-op. For example, if the `edit` action is to add a tag to a rule that already has that tag, and to delete an index pattern that doesn't exist. Skipped rules objects will only return their `id`, `name`, and `skip_reason`.

Reviewer 3: ```suggestion
Rules can only be `skipped` when performing an `edit` action, when all the operations to be performed on any one rule result in nothing being done. For example, if the `edit` action is used to add a tag to a rule that already has that tag, or to delete an index pattern that doesn't exist. Skipped rules objects will only return their `id`, `name`, and `skip_reason`.
```
